test/wasm2js/export_global.2asm.js

```diff
@@ -0,0 +1,30 @@
+
+function asmFunc(env) {
+ var Math_imul = Math.imul;
+ var Math_fround = Math.fround;
+ var Math_abs = Math.abs;
+ var Math_clz32 = Math.clz32;
+ var Math_min = Math.min;
+ var Math_max = Math.max;
+ var Math_floor = Math.floor;
+ var Math_ceil = Math.ceil;
+ var Math_trunc = Math.trunc;
+ var Math_sqrt = Math.sqrt;
+ var abort = env.abort;
+ var nan = NaN;
+ var infinity = Infinity;
+ var global0 = 655360;
+ function $0() {
+  return 42 | 0;
+ }
+ 
+ return {
+  "HELLO": global0, 
```

Comment:
I think in wasm an exported global is an object `{ value: .. }`. That allows mutable globals to be exported. If I am not wrong, then we should export such an object here.

However, I think for mutable globals we'd need more work, as we'd need to access the same object inside the module. But I think it is ok to not handle that in this PR. We can handle just the immutable case for now. (But I do think we should export an object as the wasm JS API expects.)

Reply:
if export an object instead, it will make the exports of the generated JS doesn't match the WASM. for mutable globals, I suggest generating this code:

```diff
+export var HELLO
 function asmFunc(env) {
  // ...
- var global0 = 655360;
+ HELLO = 655360;
  function $0() {
   return 42 | 0;
  }
 
  return {
-  "HELLO": global0, 
   "helloWorld": $0
  };
 }

 var retasmFunc = asmFunc(  { abort: function() { throw new Error('abort'); }
   });
-export var HELLO = retasmFunc.HELLO;
 export var helloWorld = retasmFunc.helloWorld;
```

Reply:
What do you mean by "won't match the wasm"?

IIANM, the wasm version emits an object with `.value` as described here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Global

Reply:
my wasm glue code use them as `wasm.HELLO` not `wasm.value.HELLO`, so if it exports in this way, our JS glue code will still be broken...

Reply:
oh I reinvestigated again, it do match the shape.

![image](https://user-images.githubusercontent.com/5390719/158571551-70bb9b4c-846d-486d-b83c-ce07c0f25d5c.png)

thanks!